mlflow/llama_index/pyfunc_wrapper.py

```diff
@@ -129,9 +130,11 @@ class RetrieverEngineWrapper(_LlamaIndexModelWrapperBase):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.engine_type = RETRIEVER_ENGINE_NAME
+        self.engine = self.index.as_retriever(**self.model_config)
 
-    def _build_engine_method(self) -> Callable:
-        return self.index.as_retriever(**self.model_config).retrieve
+    def _predict_single(self, *args, **kwargs) -> List[Dict]:
+        response = self.engine.retrieve(*args, **kwargs)
+        return [node.dict() for node in response]
```

Comment:
Is this call `node.dict()` what can generate a `None` value for a retriever engine type? 
Are all of the these objects serializable?